contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -143,6 +148,24 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 			applyPreimage(preimage)
 			return &h.htlcSuccessResolver, nil
 
+		case <-hodlSubscription.CancelChan:
+			// No action required, but we need to receive to prevent
+			// the channel from getting jammed.
+
+		case preimage := <-hodlSubscription.SettleChan:
+			// If this isn't our preimage, then we'll continue
+			// onwards. This should normally not happen.
+			if !preimage.Matches(h.payHash) {
```

Comment:
Non blocker: can use this above also in the `WitnessUpdates` case statement. Not yet convinced (still getting through the diff) that we actually need this as well though.

Reply:
Yes, it is a good suggestion. But I just don't want to touch more than I already do. It is a simple change, but any change incurs a slight risk. I'd rather leave it for the next who actually touches that code.